packages/flutter/lib/src/services/binding.dart

```diff
@@ -29,6 +32,20 @@ mixin ServicesBinding on BindingBase {
   static ServicesBinding get instance => _instance;
   static ServicesBinding _instance;
 
+  /// The default instance of [BinaryMessenger].
+  ///
+  /// This is used to send messages from the application to the platform, and
+  /// keeps track of which handlers have been registered on each channel so
+  /// it may dispatch incoming messages to the registered handler.
+  BinaryMessenger defaultBinaryMessenger;
+
+  /// Initializes a default [BinaryMessenger] instance that can be used for
+  /// sending platform messages.
+  @protected
```

Comment:
Is there a reason this can't be private?  Do we expect subclasses to override it to return a different binary messenger than the default?

Reply:
Ya, exactly. We'll want to set a different binary messenger for the ServicesBinding used in tests.